### PR TITLE
Refactor Expense Claim

### PR DESCRIPTION
**Pull request by shreyashah115**
#12320
Summary:

1. Removed hard-coded workflow from Expense Claim, i.e. Approval Status field.
2. Added a patch workflow_expense_claim to create a workflow which can be edited as per the need.
3. Replaced field approval_status with workflow_state.
4. Modified the test cases.
5. Tests would fail on Travis until patch is merged.

expense-workflow

expensework
workflowexpense

### CODE REVIEW

**Comment by rmehta** [comment]
@shreyashah115 is this done?

**Comment by Olawale1** [comment]
I was just about asking what is the delay in getting this merged... it'll greatly help usability of workflows in the Expense Claim document

Great job!

**Comment by rmehta** [comment]
@Olawale1 will be great if you can help by testing locally 😄 instead of just following-up

File: erpnext/patches.txt [hunk 1]
```diff
@@ -492,4 +492,9 @@ erpnext.patches.v10_0.added_extra_gst_custom_field
 erpnext.patches.v10_0.workflow_leave_application #2018-01-24
 erpnext.patches.v10_0.set_default_payment_terms_based_on_company
 erpnext.patches.v10_0.update_sales_order_link_to_purchase_order
+<<<<<<< 152bf37e7fddaf7f2902d52225ca555b79f069a3
```
Comment (nabinhait) [inline erpnext/patches.txt]: ????

File: erpnext/accounts/doctype/journal_entry/journal_entry.js [hunk 2]
```diff
@@ -116,7 +116,7 @@ erpnext.accounts.JournalEntry = frappe.ui.form.Controller.extend({
 			if(jvd.reference_type==="Expense Claim") {
 				return {
 					filters: {
-						'approval_status': 'Approved',
+						'workflow_state': 'Approved',
```
Comment (nabinhait) [inline erpnext/accounts/doctype/journal_entry/journal_entry.js]: there will be no workflow_state field in Expense Claim by default

**Review by nabinhait — CHANGES_REQUESTED** [review]
Closing this PR for now, reopen after making same changes you did for Leave Application

**Comment by Olawale1** [comment]
Hi @rmehta 

The animated gif looked good enough for me :))

Will look into testing locally when fixes are done

Thanks plenty

File: erpnext/accounts/doctype/journal_entry/journal_entry.js (hunk 2 above)
Comment (shreyashah115) [reply erpnext/accounts/doctype/journal_entry/journal_entry.js]: Was waiting for Leave Application to get merged. Will fix now.

**Pull request closed without merging**